### PR TITLE
ci: add very good gh workflow

### PR DESCRIPTION
**Pull request by adam-sroka**
This PR adds the [very good flutter GitHub workflow](https://verygood.ventures/blog/configuring-workflows-for-your-flutter-projects), refactors the whole codebase to be complient with latest linter checks, and adds a demo test for the app. Future PRs will add tests and increase the minimal code coverage threshold of the workflow.

### CODE REVIEW

**Comment by adam-sroka** [comment]
Spent some time trying to fix the [Do not use BuildContexts across async gaps](https://dart-lang.github.io/linter/lints/use_build_context_synchronously.html) warning present in all `DatePicker`s and `TimePicker`s in the app without any luck, posted a question on Flutter BLoC discord to see if I can get this moving: 
d

The easy solution would be to use an external date/time picker library instead of the native flutter material `showDatePicker(...)` ones, but ideally I'd rather avoid that and stick with defaults.

**Comment by adam-sroka** [comment]
Yesss! Finally all checks successful 🙌! Now only to get it actually working in practice! 😂

**Comment by adam-sroka** [comment]
@Archie-C maybe you want to have a look at this PR too?

File: lib/collection/view/collection_page.dart [hunk 1]
```diff
@@ -9,7 +9,7 @@ class CollectionPage extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    final List<Widget> _screens = [
+    final List<Widget> screens = [
```
Comment (Archie-C) [inline lib/collection/view/collection_page.dart:12]: is there a reason for making this not private @adam-sroka

**Pull request merged**